### DISCUSSION TITLE
Flamingo SF-501 remote

### DISCUSSION THREAD

**Discussion by bartplessers** [arendst/Tasmota#14266] [Q&A] [answered]
Hi,

Has anybody ever tried to use a flamingo sf-501 remote with sonoff rf bridge, flashed with tasmota + portisch?
![image](https://user-images.githubusercontent.com/10470708/148023673-9ae032ee-8fef-4dba-ba7f-0c0c08f30411.jpeg)

The remote was sold with RF powerplugs, but I’m only interested in the remote. It has a sleek design and 10 buttons.
A couple of years ago I bought some of them and I still haven’t found a good alternative for the remote, so I would like to re-use them in my current (zigbee + mqtt) environment.

I already tried to get codes with a sonoff rf bridge with portisch and tasmota, but no luck.

I read somewhere this has to do with ‘rolling codes’, but I’m not sure this is the problem...

What I don’t understand is that the signals of the remote easily can be cloned by a cheap Kebidu remote:
https://m.nl.aliexpress.com/item/33038462280.html

This small remote can handle the (rolling?) codes with ease, and tasmota/sonoff/portisch can't?

I’m wondering if I can ‘clone’ my sf-501 remote and receive the codes into my mqtt broker with Tasmota...
Maybe it's a problem of sonoff/portisch?
Did anyone else succeed to receive codes from a Flamingo SF-501 with other hardware?

I'm only interested in receiving the codes and translate them to mqtt. Sending RF is not important as I only want to use the remotes...

Any help is appreciated!

Kind regards,
Bart

**Comment by sfromis** [comment, marked as answer]
Tasmota+Portisch is only designed to handle simple remotes sending same simple codes every time. It would take a lot of additional complexity (code size) to handle rolling codes, and the RF chip used in the Sonoff bridge is a *very* small MCU not even having capacity for all protocols at the same time. It is possible to build it for another set of protocols, don't know if that would help getting your remote to work.

**Reply by bartplessers** [reply]
Hello @sfromis,

I can understand that there's not enough capacity on the sonoff bridge.
But how on earth than is the Kebidu capable of receiving and interpret the code of the Flamingo SF-501? It can be programmed by picking up the code of the SF-501...

And if the sonoff is not powerful enough, are you aware of other hardware that can do the job with tasmota? I have i.e some spare ESP32's on the shelf...

Kind refards,
B

**Reply by sfromis** [reply]
Well, parsing rolling codes is a routine thing to do for devices designed to emulate as many RF remotes as possible. The complexity of having very broad support takes a lot of code, and this is *neither* in the scope for Tasmora *nor* Portisch. In principle "any" feature could be added to Tasmota, as long as it can fit within a modest amount of code space and RAM, and importantly if someone is willing to do the work to implement it. And with RF 433 being a badly outdated technology, the number of hackers burning to do this may be "rather low".

To be clear, I have zero knowledge of the particulars of that remote, and hurdles in supporting it. The above was meant in general as a more abstract overview. If the issue is not rolling codes, there just *might* be a way.....

With ESP32, the code size issue is of course much less restrictive, and if someone goes to work on creating an optional and well-behaved driver for expanded RF support, I see nothing which would block if from being included in what users could build for.

**Reply by bartplessers** [reply]
Ok, thanx again @sfromis.

I will probably give OpenMQTTGateway a chance than and see if I get things working.

I know rf is outdated, and the only reason I'm looking at it is because I can't find nice (zigbee)remotes with let's say 10 buttons. I know many zigbee devices can have more than 10 actions (single/double/tripple/long click etc..) but it's unconvenient to give a whole manual to my babysit how he/she can turn on some lights...
So it's a matter of hardware/design. My house is fully equipped with zigbee lights and wall switches, and it would be much easier to stick on the zigbee track. Helas. I didn't find a good multi-purpose handheld zigbee remote yet... 

That's why I'm trying to use other technologies. 
Right now I'm looking at IR (so I can use a simple TV-remote) and RF (were my wife likes my old remotes very much 😀)

We'll see were it ends, but as I understood from your explanation: Tasmota is not a solution for my specific problem, so I will give OpenMQTTGateway a try now.


Thanx for your time,
Bart

**Reply by sfromis** [reply]
With RF, I'd look in the direction of a cheap SDR dongle and the rtl_433 open source software.

**Reply by bartplessers** [reply]
Ok, thanx for the tip.
Will take a look at that later, but first OMG. It looks that they do have a solution with the Sonoff RF bridge and a 'direct hack' (some mods on the board).  
In that case I don't need new hardware...

https://1technophile.blogspot.com/2019/04/sonoff-rf-bridge-pilight-or-how-to.html?m=1

Grtz
B

**Reply by bartplessers** [reply]
Hi again,
I'm afraid that Sonoff RF bridge is also a dead ending street. They shipped their latest releases with another (incompatible) chip...  :-)

**Comment by bartplessers** [comment]
Hi to all.
For any reference: I succeeded to flash an simple ESP8266 with OpenMQTTGateway and decode my Flamingo SF-501 remote
You can find instructions here:
https://community.openmqttgateway.com/t/flamingo-sf-501-remote/1755/2


kind regards,
Bart
